### PR TITLE
Add get_redis_version command

### PR DESCRIPTION
**Pull request by oshadmi**

### CODE REVIEW

File: src/context/mod.rs [hunk 1]
```diff
@@ -263,4 +266,54 @@ impl Context {
...
+            Ok(re) => match re.captures(s) {
+                Some(captures) => match captures.get(1) {
+                    Some(m) => Some(m.as_str()),
+                    None => None,
+                },
+                None => None,
+            },
+            Err(_) => None,
+        }
+    }
```
Comment (gavrie) [inline src/context/mod.rs:282]: No sure why you made this a method of `Context`...
In general, the version parsing is spread out over 3 functions: This one, `get_redis_version_internal` and `Version::from`.
The regex parsing also has some duplication.
What I suggest is: Write 1 function that parses the `info` response directly into a `Version`. You can use 3 capture groups to directly extract the major/minor/patch without later needing to split on `.`. This should be fairly simple and clean.

**Review by gavrie — CHANGES_REQUESTED** [review]

File: tests/integration.rs [hunk 2]
```diff
@@ -48,3 +49,29 @@ fn test_keys_pos() -> Result<()> {
...
+    let _guards = vec![start_redis_server_with_module("test_helper", port)
+        .with_context(|| "failed to start redis server")?];
+    let mut con =
+        get_redis_connection(port).with_context(|| "failed to connect to redis server")?;
+
+    let res: Vec<i64> = redis::cmd("test_helper.version")
+        .query(&mut con)
+        .with_context(|| "failed to run test_helper.version")?;
+    assert!(
+        (res[0] == 6 && res[1] == 2 && res[2] >= 3) || (res[0] == 6 && res[1] == 0 && res[2] >= 15)
```
[the remark this answers is not in the document]
Comment (oshadmi) [reply tests/integration.rs]: Fixed to test for major > 0

File: Makefile [hunk 3]
```diff
@@ -65,6 +65,7 @@ RUST_SOEXT.macos=dylib
 
 build:
 	cargo build --features experimental-api --all --all-targets  $(CARGO_FLAGS)
+	cargo build --features test --example test_helper --all-targets  $(CARGO_FLAGS)
```
Comment (gkorland) [inline Makefile:68]:
```suggestion
	cargo build --features experimental-api,test --all --all-targets  $(CARGO_FLAGS)
```

File: build.sh [hunk 4]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env sh
 cargo build --features experimental-api --all --all-targets
+cargo build --features test --example test_helper --all-targets
```
Comment (gkorland) [inline build.sh:3]:
```suggestion
cargo build --features experimental-api,test --all --all-targets
```

**Review by gkorland — CHANGES_REQUESTED** [review]

**Review by gkorland — APPROVED** [review]

**Pull request merged**